### PR TITLE
feat: add solc

### PR DESCRIPTION
**Pull request by williamboman**

### CODE REVIEW

File: lua/lspconfig/server_configurations/solc.lua [hunk 1]
```diff
@@ -0,0 +1,27 @@
+local util = require 'lspconfig.util'
+
+local bin_name = 'solc-static-linux'
+if vim.fn.has 'mac' then
+  bin_name = 'solc-macos'
+end
+if vim.fn.has 'win32' == 1 then
+  bin_name = 'solc-windows'
+end
```
Comment (williamboman) [inline lua/lspconfig/server_configurations/solc.lua:9]: This assumes installation via prebuilt binaries distributed via GitHub releases (https://github.com/ethereum/solidity/releases).

Comment (williamboman) [reply lua/lspconfig/server_configurations/solc.lua:9]: In nvim-lsp-installer these get normalized to `solc(.exe)`. I have not properly read through https://docs.soliditylang.org/en/latest/installing-solidity.html#building-from-source to know if there's a canonical name for it

File: lua/lspconfig/server_configurations/solc.lua [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+return {
+  default_config = {
+    cmd = { bin_name, '--lsp' },
+    filetypes = { 'solidity' },
+    root_dir = util.root_pattern('.git'),
```
Comment (williamboman) [inline lua/lspconfig/server_configurations/solc.lua:15]: No idea if this is a proper root_dir pattern. Copied it from `solidity_ls` and just removed `package.json`.

**Comment by williamboman** [comment]
@christianparpart Congrats on the new release 🎉 ! Do you have any feedback on this PR?

File: lua/lspconfig/server_configurations/solc.lua (hunk 1 above)
Comment (polyzen) [reply lua/lspconfig/server_configurations/solc.lua:9]: The canonical name would be `solc`:
https://docs.soliditylang.org/en/latest/using-the-compiler.html#basic-usage

**Comment by christianparpart** [comment]
> @christianparpart Congrats on the new release 🎉 ! Do you have any feedback on this PR?

Oh nice work. 
I am currently on family duty. I will have a look ASAP.

**Pull request merged**